### ISSUE TITLE
Turn Light on 

### ISSUE THREAD

**Issue by Sven9719** [Raabi91/moonraker-telegram#35]
Hello, is it possible before sending a status with image to turn on a relay that is included in Moonraker and turn it off again after recording?

**Comment by Raabi91** [comment]
hey and sorry for the late write back had privately very much around the ears. at the moment it is not possible but I can see if you can do something there

**Comment by Raabi91** [comment]
@Sven9719 
i have now two new comands in the last update
first on ist /gcode_macro if you have your leds over a gcode_macro
and the second one ist /power if you use the leds over the powerplugin

pleas let me know when it works

**Comment by Sven9719** [comment]
Hey Raabi,
the extension works perfectly, but I imagined that the LEDs would be switched on automatically before the picture is taken and then switched off again automatically afterwards.

Best regards

**Comment by Raabi91** [comment]
ah so there I have to look at how I could implement something like that run your leds via klipper or via moonraker?

**Comment by Sven9719** [comment]
They run with the moonraker Power Plugin, thank you so much 🙏

**Comment by excessnet** [comment]
@Sven9719 I didnt test it, but it should be something like this.

Make two macro for the power plugin on _printer.cfg_, something like this :

```
[gcode_macro LED_ON]
gcode:
  {action_call_remote_method("set_device_power",
                             device="led_psu",
                             state="on")}

[gcode_macro LED_OFF]
gcode:
  {action_call_remote_method("set_device_power",
                             device="led_psu",
                             state="off")}
```

then, on telegram, you just send the two commands

```
/gcode_macro LED_ON
/gcode_macro LED_OFF
```

**Comment by Sven9719** [comment]
But then I have to keep doing it myself and it doesn't happen automatically, right?

**Comment by Raabi91** [comment]
> @Sven9719 I didnt test it, but it should be something like this.
> 
> Make two macro for the power plugin on _printer.cfg_, something like this :
> 
> ```
> [gcode_macro LED_ON]
> gcode:
>   {action_call_remote_method("set_device_power",
>                              device="led_psu",
>                              state="on")}
> 
> [gcode_macro LED_OFF]
> gcode:
>   {action_call_remote_method("set_device_power",
>                              device="led_psu",
>                              state="off")}
> ```
> 
> then, on telegram, you just send the two commands
> 
> ```
> /gcode_macro LED_ON
> /gcode_macro LED_OFF
> ```
 
you only need to use /gcode_macro then you will asked with an inlinekeyboard

@Sven9719 okay i will take a look at this to do this automatically for the picture

**Comment by Raabi91** [comment]
> But then I have to keep doing it myself and it doesn't happen automatically, right?

I hope it works without any problems because unfortunately I cannot test it intensively.

Please try it out

https://github.com/Raabi91/moonraker-telegram/blob/master/docs/FAQ.md#how-to-use-automatic-led-for-cam

**Comment by digitalhandwerker** [comment]
I have a follow-up request/bug(?):

I set up LED and it is working as long as webcamd is running.
My problem is that I bound the LED and webcamd together (moonraker power plugin's `bound_service:`) so that the webcam is only running when I want to (no light, no camera needed).
But this is not working with the bot. I always get "webcam link not working image". (I tried setting the delay to 10s but still not working).

Moonraker should handle the starting and stopping of webcamd. But I think the bot is trying fetch the snapshot url before the service is running. The delay seems to start after a first fetch.
I guess this is unintended behavior?

Thanks.
